### PR TITLE
[receiver/hostmetrics]: change uptime scraper to system scraper

### PR DESCRIPTION
**Pull request by braydonk**
#### Description
This PR makes some changes that I had hoped to suggest on #35954, however I was unable to make my comment in time before the PR was merged.

The changes this PR makes are:
* The `uptime` scraper is changed to be named `system`
* The scraper is registered using the scraper name when adding to the scrapercontroller

#### Link to tracking issue
Amends the fix to #31627 (following a comment I made on #35932)

#### Documentation

The documentation that was added for the `uptime` scraper was renamed accordingly, and the description adjusted.

### CODE REVIEW

**Comment by braydonk** [comment]
I am ideally hoping that this change makes it in before a new OTel Collector is tagged so that no users start using `uptime` scrapers in their configs.

**Comment by braydonk** [comment]
I would like to skip changelog, because in this case what I did was modified the existing changelog to reflect this new change.

**Comment by braydonk** [comment]
@kevinnoel-be I am unable to officially add you as a reviewer on this PR, but if you have a second to take a look I would appreciate it!

File: receiver/hostmetricsreceiver/internal/scraper/systemscraper/factory_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package systemscraper
+
+import (
+	"context"
+	"testing"
+
+	"github.com/tj/assert"
```
Comment (kevinnoel-be) [inline receiver/hostmetricsreceiver/internal/scraper/systemscraper/factory_test.go:7]: shouldn't this be `testify/assert`?

**Comment by kevinnoel-be** [comment]
@braydonk looks good to me except for the use of one (test) module where I think it should be testify instead?

File: receiver/hostmetricsreceiver/internal/scraper/systemscraper/factory_test.go (hunk 1 above)
Comment (braydonk) [reply receiver/hostmetricsreceiver/internal/scraper/systemscraper/factory_test.go:7]: Oops thanks for catching that, `goimports` decided something totally random for me here for some reason.

Comment (braydonk) [reply receiver/hostmetricsreceiver/internal/scraper/systemscraper/factory_test.go]: Fixed in [fc04bce](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36123/commits/fc04bce64eda7b8f0bb1510550c002ac78fba649)

**Review by atoulme — APPROVED** [review]
ok, LGTM

**Review by ChrsMark — APPROVED** [review]

**Pull request merged**